### PR TITLE
Merge rn/0.70-stable hermes branch

### PR DESCRIPTION
**Pull request by vmoroz**
## Summary

This PR is merging the latest code from the rn/0.70-stable hermes branch:

- Merge tag 'hermes-2022-04-28-RNv0.69.0-15d07c2edd29a4ea0b8f15ab0588a0c1adb1200f'
- Merge tag 'hermes-2022-07-15-RNv0.70.0-88dd5731a19ab6b38b0a0a2d4386ba959f2a2c98'
- Merge tag 'hermes-2022-07-20-RNv0.70.0-bc97c5399e0789c0a323f8e1431986e207a9e8ba'
- Update react-native dependency to v0.70.0-rc.3
- Update package.json dependencies for NAPI tests
- Change hermes-windows version to 0.12.3

## Test Plan

The code passes all unit tests

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/hermes-windows/pull/118)

### CODE REVIEW

File: lib/Support/OSCompatWindows.cpp [hunk 1]
```diff
@@ -407,6 +422,18 @@ int sched_getcpu() {
   return -1;
 }
 
+uint64_t cpu_cycle_counter() {
+#if defined(_MSC_VER) && (defined(_M_IX86) || defined(_M_X64))
+  return __rdtsc();
```
Comment (jonthysell) [inline lib/Support/OSCompatWindows.cpp:427]: Totally optional, but if you ever wanted ARM32 to build:

```suggestion
  return __rdtsc();
#elif defined(_MSC_VER) && defined(_M_ARM)
  return __rdpmccntr64();
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -134,6 +127,15 @@ set(HERMESVM_INDIRECT_THREADING ${DEFAULT_INTERPRETER_THREADING} CACHE BOOL
 set(HERMESVM_ALLOW_COMPRESSED_POINTERS ON CACHE BOOL
   "Enable compressed pointers. If this is on and the target is a 64-bit build, compressed pointers will be used.")
 
+if(APPLE OR WIN32)
+    set(DEFAULT_CONTIGUOUS_HEAP OFF)
```
Comment (aeulitz) [inline CMakeLists.txt:131]: [nit] Comment why this is needed?

File: CMakeLists.txt [hunk 3]
```diff
@@ -626,7 +648,7 @@ if(HERMES_IS_ANDROID)
   find_package(fbjni REQUIRED CONFIG)
 endif()
 
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
```
Comment (aeulitz) [inline CMakeLists.txt:651]: > 17 [](http://example.com/codeflow?start=22&length=3)

OOC, is that in sync with the fb hermes repo?

Comment (aeulitz) [reply CMakeLists.txt:651]: To answer my own question: Yes, it is.

File: CMakeLists.txt (hunk 2 above)
Comment (aeulitz) [reply CMakeLists.txt:131]: If I understand correctly, this change is an "integrate" of the changes in fb hermes. As such, we shouldn't add anything here that isn't in fb hermes. If this is correct, feel free to resolve this comment without further actions.

**Review by aeulitz — APPROVED** [review]
:shipit:

File: CMakeLists.txt (hunk 3 above)
Comment (vmoroz) [reply CMakeLists.txt:651]: Yes, they switched to C++17.

File: lib/Support/OSCompatWindows.cpp (hunk 1 above)
Comment (vmoroz) [reply lib/Support/OSCompatWindows.cpp:427]: Thanks for the suggestion! We do not compile this code for ARM, so I am hesitant to include it in this PR. We can do it in future if we ever need to target ARM (not ARM64).

File: CMakeLists.txt (hunk 2 above)
Comment (vmoroz) [reply CMakeLists.txt:131]: Yes, this code came from Meta.

**Pull request merged**